File: api/java/editor/api-editor/src/main/java/ca/bc/gov/tno/areas/editor/controllers/SourceMetricController.java
package ca.bc.gov.tno.areas.editor.controllers;

import java.util.List;

import javax.annotation.security.RolesAllowed;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import ca.bc.gov.tno.dal.db.services.interfaces.ISourceMetricService;
import ca.bc.gov.tno.areas.editor.models.SourceMetricModel;

/**
 * SourceMetricController class, provides endpoints for metrics.
 */
@RolesAllowed({ "administrator", "editor" })
@RestController("EditorSourceMetricController")
@RequestMapping({ "/editor/source/metrics", "/api/editor/source/metrics" })
public class SourceMetricController {

  /**
   * DAL for metric.
   */
  private ISourceMetricService metricService;

  /**
   * Creates a new instance of a SourceMetricController object, initializes with
   * specified parameters.
   *
   * @param metricService SourceMetric service.
   */
  @Autowired
  public SourceMetricController(ISourceMetricService metricService) {
    this.metricService = metricService;
  }

  /**
   * Request a list of all metrics from the db.
   *
   * @return
   */
  @GetMapping(path = { "", "/" }, produces = MediaType.APPLICATION_JSON_VALUE)
  public List<SourceMetricModel> findAll() {
    var results = metricService.findAll();
    var models = results.stream().map(a -> new SourceMetricModel(a)).toList();
    return models;
  }

}

File: api/net/Areas/Editor/Controllers/SourceMetricController.cs
using System.Net;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;
using TNO.API.Areas.Editor.Models.SourceMetric;
using TNO.API.Models;
using TNO.DAL.Services;

namespace TNO.API.Areas.Editor.Controllers;

/// <summary>
/// SourceMetricController class, provides SourceMetric endpoints for the api.
/// </summary>
[Authorize]
[ApiController]
[Area("editor")]
[ApiVersion("1.0")]
[Route("api/v{version:apiVersion}/[area]/source/metrics")]
[Route("api/[area]/source/metrics")]
[Route("v{version:apiVersion}/[area]/source/metrics")]
[Route("[area]/source/metrics")]
[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Unauthorized)]
[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Forbidden)]
public class SourceMetricController : ControllerBase
{
    #region Variables
    private readonly ISourceMetricService _service;
    #endregion

    #region Constructors
    /// <summary>
    /// Creates a new instance of a SourceMetricController object, initializes with specified parameters.
    /// </summary>
    /// <param name="service"></param>
    public SourceMetricController(ISourceMetricService service)
    {
        _service = service;
    }
    #endregion

    #region Endpoints
    /// <summary>
    /// Return an array of SourceMetric.
    /// </summary>
    /// <returns></returns>
    [HttpGet]
    [Produces("application/json")]
    [ProducesResponseType(typeof(IEnumerable<SourceMetricModel>), (int)HttpStatusCode.OK)]
    [SwaggerOperation(Tags = new[] { "SourceMetric" })]
    public IActionResult FindAll()
    {
        return new JsonResult(_service.FindAll().Select(c => new SourceMetricModel(c)));
    }
    #endregion
}
